homeassistant/components/intent_script.py

```diff
@@ -76,8 +76,12 @@ def async_handle(self, intent_obj):
         card = self.config.get(CONF_CARD)
         action = self.config.get(CONF_ACTION)
         is_async_action = self.config.get(CONF_ASYNC_ACTION)
-        slots = {key: value['value'] for key, value
-                 in intent_obj.slots.items()}
+        slots = {}
+        for key, value in intent_obj.slots.items():
+            if value.get('value') is not None:
```

Comment:
We should not add any magic in how we handle slots. Instead we should just change it from `slots = {}` to be

```python
vars = {
  slots: intent_obj.slots
}

for key, value in slots.items():
    vars[key] = value['value']
```

Reply:
The google and alexa components only pass a dict like this, so there is none of the extra information.

{
  'slotName': 'mySlot'
  'value': 'myValue' ~<--- this is slots.value.value from the original intent
}
My change simply adds a slot called 'slots' with the extra info. I've jumped through these hoops only to not introduce a breaking change.

Would you be opposed to me redoing this pull request as simply refactoring the Intent object from 
```    def __init__(self, hass, platform, intent_type, slots, text_input):
to
```    def __init__(self, hass, platform, intent):

This will end up being much cleaner in the end without these kludges and will be easy to still support current intents. Just want to see if you are ok with that before I put the work in.

Reply:
It doesn't make sense to pass the intent object to templates. Templates are for rendering responses. If you are registering an intent handler in Python, you already get the intent object and can do whatever you want. 

I am starting to doubt now if we really should expose everything to the template or that you're trying to use templates for something that it is not supposed to do.

Can you give me an example of how you would use extended slot information in a template?

Reply:
TL; DR; how about instead we parse the specific snips builtin slot types in the snips component to set a useful value, ie "75" instead of seventy five"? No need to change anything other than the snips component.

So in the specific intent in the description, timer_duration = "five minutes" which ends up needing to be parsed out to get it in a value where I can set a delay for it. But the data is already parsed in the extended slot info (minutes=5)

Since this specific case is in reference to only the snips builtin slot type of duration (there are others as well like temperature). These parse out the numerical values from the spoken value. Obviously useful since HASS can't set the thermostat to "seventy five" 

Since this is specific to snips and is a builtin feature of their assistant, it probably makes more sense to just modify snips.py and call it done. Does that make sense?

Reply:
Yes that makes sense